src/agent/src/config.rs

```diff
@@ -570,6 +583,7 @@ mod tests {
             no_proxy: &'a str,
             guest_components_rest_api: GuestComponentsFeatures,
             guest_components_procs: GuestComponentsProcs,
+            image_registry_auth: &'a str,
```

Comment:
Since the field is under a `#[cfg(feature = "guest-pull")]`, don't you need the same on that code as well?

Reply:
The 4 initial commits were borrowed from #9904 actually. They will be removed at some point.